File: DataCleaningScripts/cad_filter_05.pl
### filter cad file

open $readfh,'<',"data/cad.add.160614.website.txt" or die "$!";
open $outfh, '>', "data/cad_p05_chrpos.txt" or die "$!";

$header = "SNP\tbeta\tse\teaf\teffect_allele\tother_allele\tpval\tchr\tposn";

print $outfh $header;
$trash = <$readfh>;

while(<$readfh>)
{
$_ =~ s/\./0\./g;

@ls = split(/\s+/, $_);
#print "@ls\n";
$pval = $ls[10];

if($pval < 0.05)
{
#print "$pval\n";
print $outfh "$ls[0]\t$ls[8]\t$ls[9]\t$ls[5]\t$ls[3]\t$ls[4]\t$ls[10]\t$ls[1]\t$ls[2]\n";
}
}

File: DataCleaningScripts/cad_filter_05.R

rm(list=ls())

setwd("C:/Users/cwardcav/Desktop/GitHub/MR_BACOn/DataCleaningScripts/data/")
vv <- read.table(file="cad.add.160614.website.txt", 
                 header=TRUE, stringsAsFactors = FALSE, sep="\t", fill=TRUE, quote="")
vv <- vv[,c("markername","beta","se_dgc","effect_allele_freq","effect_allele","noneffect_allele","p_dgc")]
names(vv) <- c("SNP","beta","se","eaf","effect_allele","other_allele","pval")

write.table(vv, file="cad_p05_R.txt",col.names=TRUE, row.names=FALSE, sep="\t", quote=FALSE)